### PR TITLE
internal/envoy: Set XffNumTrustedHops to keep header proto intact

### PR DESCRIPTION
**Pull request by stevesloka**
If a user has an external load balancer that terminates TLS, the X-Forwarded-Proto
header gets overwritten unless the downstream connection is trusted.

This adds a config file option to set the number of trusted hops which will
allow the headers to be instact already set from downstream.

Fixes https://github.com/projectcontour/contour/issues/3294

Signed-off-by: Steve Sloka <slokas@vmware.com>

### CODE REVIEW

**Comment by stevesloka** [comment]
This still needs docs & feature tests.

File: pkg/config/parameters_test.go [hunk 1]
```diff
@@ -86,7 +86,7 @@ default-http-versions: []
 cluster:
   dns-lookup-family: auto
 `
-	assert.Equal(t, strings.TrimSpace(string(data)), strings.TrimSpace(expected))
+	assert.Equal(t, strings.TrimSpace(expected), strings.TrimSpace(string(data)))
```
Comment (stevesloka) [inline pkg/config/parameters_test.go:89]: Debugging tests I noticed this was swapped.

**Review by youngnick — APPROVED** [review]
LGTM

**Review by sunjayBhatia — COMMENTED** [review]
LGTM pending an addition I’m guessing you’re going to add here https://github.com/projectcontour/contour/blob/main/site/docs/main/configuration.md

**Comment by stevesloka** [comment]
@sunjayBhatia yup added docs and a feature test to cover.

File: site/docs/main/configuration.md [hunk 2]
```diff
@@ -106,6 +107,16 @@ The cluster configuration block can be used to configure various parameters for
 {: class="table thead-dark table-bordered"}
 <br>
 
+### Network Configuration
```
Comment (sunjayBhatia) [inline site/docs/main/configuration.md:110]: Do we have any issues we can earmark so we make sure to add configuration to this object?

**Review by sunjayBhatia — APPROVED** [review]
LGTM, just a tiny discussion starter comment

File: site/docs/main/configuration.md (hunk 2 above)
Comment (stevesloka) [reply site/docs/main/configuration.md:110]: There's an issue linked for the max hops, are you referring to the `Network Configuration` section?

Comment (sunjayBhatia) [reply site/docs/main/configuration.md:110]: Yeah, just wondering if we already had other things to add in the `network` hash we already know about

Comment (stevesloka) [reply site/docs/main/configuration.md:110]: Yeah at the moment not sure, things that the chunked length maybe should have gone into there. We can remove the `network` parent, but was trying to group stuff together in a way.

Comment (youngnick) [reply site/docs/main/configuration.md:110]: I think having the `network` top-level item is great, we should definitely keep that, and look out for things to put in there.

Comment (stevesloka) [reply site/docs/main/configuration.md:110]: 👍

**Pull request merged**